### PR TITLE
Resolve injected changes to the root project's directory

### PR DESCRIPTION
**Pull request by lptr**
This way we can pass relative paths from Gradle profiler when --project-dir is used. Otherwise the files would be resolved based on the working directory of Gradle profiler itself.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java [hunk 1]
```diff
@@ -264,7 +277,8 @@ public void beforeOutputChange() {
 
                     @Override
                     public void beforeOutputChange(Iterable<String> affectedOutputPaths) {
-                        routingVirtualFileSystem.update(affectedOutputPaths, () -> {});
+                        routingVirtualFileSystem.update(affectedOutputPaths, () -> {
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Why did you add the linebreak here? I think it was better readable before.

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java [hunk 2]
```diff
@@ -162,13 +165,23 @@ VirtualFileSystem createVirtualFileSystem(
                 listenerManager.addListener(new RootBuildLifecycleListener() {
                     @Override
                     public void afterStart(GradleInternal gradle) {
-                        Map<String, String> systemPropertiesArgs = gradle.getStartParameter().getSystemPropertiesArgs();
+                        StartParameter startParameter = gradle.getStartParameter();
+                        Map<String, String> systemPropertiesArgs = startParameter.getSystemPropertiesArgs();
                         if (isRetentionEnabled(systemPropertiesArgs)) {
-                            Iterable<String> changedPathsSinceLastBuild = getChangedPathsSinceLastBuild(systemPropertiesArgs);
-                            for (String changedPathSinceLastBuild : changedPathsSinceLastBuild) {
+                            FileResolver fileResolver = new BaseDirFileResolver(startParameter.getCurrentDir(), () -> {
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: I probably wouldn't have used a `FileResolver` here, but a simple `File`. Then the relative paths would be converted into the absolute paths by using `new File(currentDir, relativePath).getAbsolutePath()`.

Why do you want to use a `FileResolver` here?

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Accident, I reformatted the whole file and didn't notice that this should have been reverted.

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java (hunk 2 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Because we might want to pass in absolute files as well, and then I'd have had to add a check whether the path is absolute. And we already have that logic in the `FileResolver`.

All this code will come out as soon as we have some sort of file watching. So even if this is not the most optimal I think it's fine.

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Fixed now.

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java (hunk 2 above)
Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Why do you need a check for absolute paths? `new File(baseDir, absolutePath)` already returns the correct result, aka pointing to the `absolutePath`, doesn't it?

Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: It doesn't.
